modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/rest/SearchPaginatedAPIsWithMultipleStatusTestCase.java

```diff
@@ -92,11 +92,19 @@ public void testPaginationWithMultipleStatus() throws Exception {
             Thread.sleep(500);
         }
 
-        Thread.sleep(10000);//Wait till APIs get indexed
-        HttpResponse response = apiStore.searchPaginateAPIs(TENANT_DOMAIN, "0", "10", API_NAME_PREFIX);
-        JSONObject responseJSON = new JSONObject(response.getData());
-        JSONArray returnedAPIs = responseJSON.getJSONArray("result");
-        Assert.assertEquals(returnedAPIs.length(), PAGINATED_COUNT);
+        //Wait till APIs get indexed
+        int returnApiCount = 0;
+        for (int i = 0; i < 25; i++) {
+            HttpResponse response = apiStore.searchPaginateAPIs(TENANT_DOMAIN, "0", "10", API_NAME_PREFIX);
+            JSONObject responseJSON = new JSONObject(response.getData());
+            JSONArray returnedAPIs = responseJSON.getJSONArray("result");
+            returnApiCount = returnedAPIs.length();
+            if (returnApiCount == PAGINATED_COUNT) {
+                break;
+            }
+            Thread.sleep(5000);
```

Comment:
Use awaitability library than thread.sleep.

Reply:
Currently this library is not available in our product, since this is an improvement across the product it, it will require a huge effort therefore we will fix it in the future, I created a issue [1] to track this [1] 

[1] - https://github.com/wso2/product-apim/issues/5204